File: test/test/UIViewController+extension.m




//
//  UIViewController+extension.m
//  test
//
//  Created by lingda on 2019/5/30.
//  Copyright © 2019年 lingda. All rights reserved.
//

#import "UIViewController+extension.h"
#import <objc/runtime.h>

@implementation UIViewController (extension)
+ (void)load {
    Method oriMethod = class_getInstanceMethod([self class], NSSelectorFromString(@"dealloc"));
    Method repMethod = class_getInstanceMethod([self class], @selector(replaceDelloc));
    method_exchangeImplementations(oriMethod, repMethod);
}

- (void)replaceDelloc {
    NSLog(@"d");
}
@end

File: SwiftRuntime/SwiftRuntime/Swizzle/UIViewController+Extension.swift
//
//  UIViewController+Extension.swift
//  SwiftRuntime
//
//  Created by lingda on 2019/5/28.
//  Copyright © 2019年 lingda. All rights reserved.
//

import UIKit
extension ViewController: SwizzleProtocol {
    static func swizzle() {
        swizzle(origin: #selector(viewDidLoad), des: #selector(ld_viewDidLoad))
    }
    @objc func ld_viewDidLoad() {
        ld_viewDidLoad()
        print("替换了")
    }
}
